<launch>
    <!-- USB CAMERA -->
    <node name="usb_cam" pkg="usb_cam" type="usb_cam_node" output="screen">
        <param name="video_device" value="/dev/video0" />
        <param name="image_width" value="640" />
        <param name="image_height" value="480" />
        <param name="pixel_format" value="yuyv" />
        <param name="camera_frame_id" value="usb_cam" />
        <param name="io_method" value="mmap"/>
        <param name="camera_info_url" value="package://usb_cam/camera_info/head_camera.yaml"/>
    </node>

    <!-- DETECTNET -->
    <arg name="model_name" default="ssd-mobilenet-v2"/>
    <arg name="model_path" default=""/>
    <arg name="prototxt_path" default=""/>
    <arg name="class_labels_path" default=""/>
    <arg name="input_blob" default=""/>
    <arg name="output_cvg" default=""/>
    <arg name="output_bbox" default=""/>
    <arg name="overlay_flags" default="box,labels,conf"/>
    <arg name="mean_pixel_value" default="0.0"/>
    <arg name="threshold" default="0.5"/>

    <node pkg="ros_deep_learning" type="detectnet" name="detectnet" output="screen">
        <remap from="/detectnet/image_in" to="/usb_cam/image_raw"/>
        <param name="model_name" value="$(arg model_name)"/>
        <param name="model_path" value="$(arg model_path)"/>
        <param name="prototxt_path" value="$(arg prototxt_path)"/>
        <param name="class_labels_path" value="$(arg class_labels_path)"/>
        <param name="input_blob" value="$(arg input_blob)"/>
        <param name="output_cvg" value="$(arg output_cvg)"/>
        <param name="output_bbox" value="$(arg output_bbox)"/>
        <param name="overlay_flags" value="$(arg overlay_flags)"/>
        <param name="mean_pixel_value" value="$(arg mean_pixel_value)"/>
        <param name="threshold" value="$(arg threshold)"/>
    </node>

    <!-- VIDEO OUTPUT -->
    <arg name="output" default="display://0"/>
    <arg name="output_codec" default="unknown"/>
    <arg name="output_bitrate" default="0"/>
    <include file="$(find ros_deep_learning)/launch/video_output.ros1.launch">
        <arg name="topic" value="/detectnet/overlay"/>
        <arg name="output" value="$(arg output)"/>
        <arg name="output_codec" value="$(arg output_codec)"/>
        <arg name="output_bitrate" value="$(arg output_bitrate)"/>
    </include>

    <!-- APRILTAG -->
    <node pkg="apriltag_ros" type="apriltag_ros_continuous_node" name="apriltag_ros_continuous_node" output="screen">
        <remap from="image_rect" to="/usb_cam/image_raw" />
        <remap from="camera_info" to="/usb_cam/camera_info" />
        <param name="publish_tag_detections_image" type="bool" value="true" />
        <rosparam command="load" file="$(find detection)/config/settings.yaml"/>
        <rosparam command="load" file="$(find detection)/config/tags.yaml"/>
    </node>

    <!-- CUSTOM APRILTAG LISTENER -->
    <node pkg="detection" type="detect.py" name="detect" output="screen"> 
    </node>

    <node pkg="nodelet" type="nodelet" name="szar_ultrasonic"
          args="load AREBot_ROS/UltrasonicNode arebot_nodelet_manager">
        <rosparam file="$(find szar_ultrasonic)/params/sensors.yaml"/>
    </node>

    <node pkg="nodelet" type="nodelet" name="szarbot_arm"
          args="load SZARBot_ROS/ArmNode arebot_nodelet_manager">
        <rosparam file="$(find szarbot_arm)/params/arm.yaml"/>
    </node>

</launch>

